### PR TITLE
transform: add --out flag

### PR DESCRIPTION
**Pull request by yoshuawuyts**
Implements #34 

Adds file exorcising through a `--out` / `-o` flag. All added logic takes place in `./transform.js`, the rest are just support files
- [x] exorcise files
- [x] write tests
- [x] write docs
- [x] write to stream if is stream

### CODE REVIEW

**Comment by yoshuawuyts** [comment]
PR is all ready, file exorcising works!

**Comment by yoshuawuyts** [comment]
Oh crap, actually we should also be able to pass in a stream from the JS API so servers like [bankai](https://www.npmjs.com/package/bankai) will be able to parse it happily. Adding another bullet point!

**Comment by yoshuawuyts** [comment]
Patch should be all good now!

**Comment by yoshuawuyts** [comment]
Merging

File: transform.js [hunk 1]
```diff
@@ -39,23 +50,40 @@ function transform (filename, opts) {
...
+          // exorcise to external file
+          const dirname = path.dirname(opts.out)
+          mkdirp(dirname, function (err) {
+            if (err) return done(err)
+            const ws = (typeof opts.out === 'string')
+              ? fs.createWriteStream(opts.out)
+              : opts.out
+            eos(ws, done)
+            node.update('0')
+            ws.end(css)
```
Comment (ahdinosaur) [inline transform.js]: does this work when using browserify to bundle multiple files? from my understanding, the transform will be created once for every file being transformed, which means a new writeable stream will be created for every file in the bundle. [i came up against this](https://github.com/sheetify/sheetify/issues/34#issuecomment-172820982) when experimenting with [sheetify/bundle-css](https://github.com/sheetify/bundle-cssify).
which is why [i went to the directory approach](https://github.com/sheetify/bundle-cssify/commit/22ac32c80fafdf39d73573421f2877775c224473).

**Comment by ahdinosaur** [comment]
crushing it :fireworks:

File: transform.js (hunk 1 above)
Comment (yoshuawuyts) [inline transform.js]: yeah, that would be correct - but given that to my understanding the transform happens in sequence (e.g. next chunk isn't parsed until this one is done) that should not be an issue... I hope?

**Pull request closed without merging**